JSDemos/Demos/Tabs/Overview/React/App.tsx

```diff
@@ -43,70 +60,99 @@ const App = () => {
     setIconPosition(e.value);
   }, [setIconPosition]);
 
-  const orientationChanged = React.useCallback(
-    (e) => {
-      setWidgetWrapperClasses(`widget-wrapper widget-wrapper-${e.value}`);
-      setOrientation(e.value);
-    }, [setOrientation, setWidgetWrapperClasses],
-  );
+  const orientationChanged = React.useCallback((e) => {
+    setWidgetWrapperClasses(`widget-wrapper widget-wrapper-${e.value}`);
+    setOrientation(e.value);
+  }, [setOrientation, setWidgetWrapperClasses]);
 
   const showNavigationChanged = React.useCallback((e) => {
+    const shouldRestrictWidth = e.value || scrollContent;
+
+    toggleShouldRestrictWidth(shouldRestrictWidth);
     setShowNavigation(e.value);
-  }, [setShowNavigation]);
+  }, [scrollContent, setShowNavigation, toggleShouldRestrictWidth]);
 
   const scrollContentChanged = React.useCallback((e) => {
+    const shouldRestrictWidth = e.value || showNavigation;
+
+    toggleShouldRestrictWidth(shouldRestrictWidth);
     setScrollContent(e.value);
-  }, [setScrollContent]);
+  }, [showNavigation, setScrollContent, toggleShouldRestrictWidth]);
 
   const fullWidthChanged = React.useCallback((e) => {
     setFullWidth(e.value);
-  }, [setFullWidth]);
+    setWidth(e.value ? '100%' : 'auto');
+  }, [setFullWidth, setWidth]);
 
   const rtlEnabledChanged = React.useCallback((e) => {
     setRtlEnabled(e.value);
   }, [setRtlEnabled]);
 
+  const selectionChanged1 = React.useCallback((args) => {
+    if (args.name === 'selectedIndex') {
+      setSelectedIndex1(args.value);
+    }
+  },
+  [setSelectedIndex1]);
+
+  const selectionChanged2 = React.useCallback((args) => {
+    if (args.name === 'selectedIndex') {
+      setSelectedIndex2(args.value);
+    }
+  },
+  [setSelectedIndex2]);
```

Comment:
I suggest that using post index in names of variables can confuse. Such names do not provide information about what exactly they are responsible for.

Reply:
These methods was removed in [5b9bd3875eddb26acdca83ce94cfe7a767786467].